### PR TITLE
logs.0.6.3: fix incorrect constraint on cmdliner

### PR DESCRIPTION
**Pull request by mseri**
Signed-off-by: Marcello Seri <marcello.seri@gmail.com>

### CODE REVIEW

**Comment by mseri** [comment]
Seen on: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2b4d2410715281158ca61c55fdbb853e67b3cb2d/variant/opam-2.1,compilers,4.04,cohttp-lwt-jsoo.2.5.6,lower-bounds

**Comment by kit-ty-kate** [comment]
For posterity (urls won’t stay available for long sadly):
```
#=== ERROR while compiling logs.0.6.3 =========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.04.2 | file:///home/opam/opam-repository
# path                 ~/.opam/4.04/.opam-switch/build/logs.0.6.3
# command              ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml build --pinned false --with-js_of_ocaml true --with-fmt false --with-cmdliner true --with-lwt true
# exit-code            1
# env-file             ~/.opam/log/logs-24-f0ea39.env
# output-file          ~/.opam/log/logs-24-f0ea39.out
### output ###
# ocamlfind ocamldep -modules src/logs.ml > src/logs.ml.depends
# ocamlfind ocamldep -modules src/logs.mli > src/logs.mli.depends
# ocamlfind ocamlc -c -bin-annot -safe-string -I src -I test -o src/logs.cmi src/logs.mli
# ocamlfind ocamlopt -c -bin-annot -safe-string -I src -I test -o src/logs.cmx src/logs.ml
# ocamlfind ocamlopt -a src/logs.cmx -o src/logs.cmxa
# ocamlfind ocamlopt -shared src/logs.cmx -o src/logs.cmxs
# ocamlfind ocamlc -c -bin-annot -safe-string -I src -I test -o src/logs.cmo src/logs.ml
# ocamlfind ocamlc -a src/logs.cmo -o src/logs.cma
# ocamlfind ocamldep -package js_of_ocaml -modules src/logs_browser.ml > src/logs_browser.ml.depends
# ocamlfind ocamldep -package js_of_ocaml -modules src/logs_browser.mli > src/logs_browser.mli.depends
# ocamlfind ocamlc -c -bin-annot -safe-string -package js_of_ocaml -I src -I test -o src/logs_browser.cmi src/logs_browser.mli
# ocamlfind ocamlopt -c -bin-annot -safe-string -package js_of_ocaml -I src -I test -o src/logs_browser.cmx src/logs_browser.ml
# ocamlfind ocamlopt -a -package js_of_ocaml src/logs_browser.cmx -o src/logs_browser.cmxa
# ocamlfind ocamlopt -shared -package js_of_ocaml src/logs_browser.cmx -o src/logs_browser.cmxs
# ocamlfind ocamlc -c -bin-annot -safe-string -package js_of_ocaml -I src -I test -o src/logs_browser.cmo src/logs_browser.ml
# ocamlfind ocamlc -a -package js_of_ocaml src/logs_browser.cmo -o src/logs_browser.cma
# ocamlfind ocamldep -package cmdliner -modules src/logs_cli.ml > src/logs_cli.ml.depends
# ocamlfind ocamldep -package cmdliner -modules src/logs_cli.mli > src/logs_cli.mli.depends
# ocamlfind ocamlc -c -bin-annot -safe-string -package cmdliner -I src -I test -o src/logs_cli.cmi src/logs_cli.mli
# + ocamlfind ocamlc -c -bin-annot -safe-string -package cmdliner -I src -I test -o src/logs_cli.cmi src/logs_cli.mli
# File "src/logs_cli.mli", line 15, characters 17-33:
# Error: Unbound type constructor Cmdliner.Arg.env
# Command exited with code 2.
# pkg.ml: [ERROR] cmd ["ocamlbuild"; "-use-ocamlfind"; "-classic-display"; "-build-dir";
#      "_build"; "opam"; "pkg/META"; "CHANGES.md"; "LICENSE.md"; "README.md";
#      "src/logs.a"; "src/logs.cmxs"; "src/logs.cmxa"; "src/logs.cma";
#      "src/logs.cmx"; "src/logs.cmi"; "src/logs.mli"; "src/logs_browser.a";
#      "src/logs_browser.cmxs"; "src/logs_browser.cmxa";
#      "src/logs_browser.cma"; "src/logs_browser.cmx"; "src/logs_browser.cmi";
#      "src/logs_browser.mli"; "src/logs_cli.a"; "src/logs_cli.cmxs";
#      "src/logs_cli.cmxa"; "src/logs_cli.cma"; "src/logs_cli.cmx";
#      "src/logs_cli.cmi"; "src/logs_cli.mli"; "src/logs_lwt.a";
#      "src/logs_lwt.cmxs"; "src/logs_lwt.cmxa"; "src/logs_lwt.cma";
#      "src/logs_lwt.cmx"; "src/logs_lwt.cmi"; "src/logs_lwt.mli";
#      "src/logs_top_init.ml"; "src/logs_fmt_top_init.ml"; "test/tool.ml";
#      "test/tags.ml"]: exited with 10
```

**Comment by mseri** [comment]
I was just about posting that :D

File: packages/logs/logs.0.6.3/opam [hunk 1]
```diff
@@ -16,10 +16,12 @@ depends: [
 depopts: [
   "js_of_ocaml"
   "fmt"
-  "cmdliner" {>= "1.0.4."}
+  "cmdliner"
   "lwt" ]
 conflicts: [
-  "js_of_ocaml" { < "3.3.0" } ]
+  "js_of_ocaml" { < "3.3.0" } 
+  "cmdliner" { < "1.0.0" }
```
Comment (kit-ty-kate) [inline packages/logs/logs.0.6.3/opam:23]:
```suggestion
  "cmdliner" { < "0.9.8" }
```
`Cmdliner.Arg.env` was added in 0.9.8: https://github.com/dbuenzli/cmdliner/commit/e82fd02842719167c3a04fc7ed0edf9a8036203c

Comment (kit-ty-kate) [reply packages/logs/logs.0.6.3/opam:23]:
```
-> installed ocamlfind.1.9.1
-> installed ocamlbuild.0.14.0
-> installed cmdliner.0.9.8
-> installed topkg.1.0.4
-> installed logs.0.6.3
```
Tested on OCaml 4.05. It seems to work fine.

**Pull request merged**